Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -56,25 +63,291 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(CookieStore);
 
-Ref<CookieStore> CookieStore::create(Document* document)
+class CookieStore::MainThreadBridge : public ThreadSafeRefCounted<MainThreadBridge, WTF::DestructionThread::Main> {
+public:
+    static Ref<MainThreadBridge> create(CookieStore& cookieStore)
+    {
+        return adoptRef(*new MainThreadBridge(cookieStore));
+    }
+
+    void get(CookieStoreGetOptions&&, int promiseIdentifier);
+    void getAll(CookieStoreGetOptions&&, int promiseIdentifier, SecurityOrigin*, bool globalObjectIsWindowObject);
+    void set(CookieInit&& options, int promiseIdentifier);
+
+private:
+    MainThreadBridge(CookieStore&);
+
+    void ensureOnMainThread(Function<void(ScriptExecutionContext& context)>&&);
+    void ensureOnContextThread(ScriptExecutionContextIdentifier, Function<void()>&&);
+
+    WeakPtr<CookieStore, WeakPtrImplWithEventTargetData> m_cookieStore;
+};
+
+CookieStore::MainThreadBridge::MainThreadBridge(CookieStore& cookieStore)
+    : m_cookieStore(cookieStore)
```

Comment:
We probably need to store the identifier of the cookieStore's context as a data member too, so that we can do ensureOnContextThread() properly